kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/PaginationContext.java

```diff
@@ -23,13 +25,20 @@
 /** {@code PaginationContext} carries pagination-related information. */
 public class PaginationContext {
 
-  public static PaginationContext forPageWithPageToken(long pageSize, PageToken pageToken) {
+  public static PaginationContext forPageWithPageToken(
+      long pageSize, PageToken pageToken, String tablePath, long tableVersion) {
     Objects.requireNonNull(pageToken, "page token is null");
-    return new PaginationContext(pageSize, Optional.of(pageToken));
+    Objects.requireNonNull(tablePath, "table Path is null");
+    checkArgument(tablePath.equals(pageToken.getTablePath()), "table path changes!");
```

Comment:
can we throw a clearer error message here?

i.e. the X in the pageToken does not match the X that was taken from the ScanBuilder/Scan context -- i.e. you probably passed in the incorrect pageToken?